tests/Node/SetTest.php

```diff
@@ -77,7 +77,7 @@ public static function provideTests(): iterable
         $node = new SetNode(true, $names, $values, 1);
         $tests[] = [$node, <<<EOF
 // line 1
-\$context["foo"] = ('' === \$tmp = "foo") ? '' : new Markup(\$tmp, \$this->env->getCharset());
+\$context["foo"] = new Markup("foo", \$this->env->getCharset());
```

Comment:
shouldn't we also have a test covering the case of non-constant expressions ?

Reply:
We already have such tests above, right?